app/controllers/api/v1/conversations_controller.rb

```diff
@@ -14,40 +14,42 @@ class ConversationsController < Api::V1::BaseController
       end
 
       rescue_from ActiveRecord::RecordNotFound do
-        render(
-          json:   {error: I18n.t("conversations.not_found")},
-          status: :not_found
-        )
+        render json: I18n.t("api.endpoint_errors.conversations.not_found"), status: :not_found
       end
 
       def index
-        params.permit(:only_after, :unread)
-        conversations = conversation_service.all_for_user(params)
+        params.permit(:only_after, :only_unread)
+        mapped_params = {}
+        mapped_params[:only_after] = params[:only_after] if params.has_key?(:only_after)
+
+        mapped_params[:unread] = params[:only_unread] if params.has_key?(:only_unread)
+        conversations = conversation_service.all_for_user(mapped_params)
         render json: conversations.map {|x| conversation_as_json(x) }
       end
 
       def show
         conversation = conversation_service.find!(params[:id])
-        render json: {
-          conversation: conversation_as_json(conversation)
-        }
+        render json: conversation_as_json(conversation)
       end
 
       def create
+        params.require(%i[subject body recipients])
+        recipient_ids = JSON.parse(params[:recipients]).map {|p| Person.find_from_guid_or_username(id: p).id }
         conversation = conversation_service.build(
           params[:subject],
           params[:body],
-          params[:recipients]
+          recipient_ids
         )
+        raise ActiveRecord::RecordInvalid unless conversation.participants.length == (recipient_ids.length + 1)
         conversation.save!
         Diaspora::Federation::Dispatcher.defer_dispatch(
           current_user,
           conversation
         )
 
-        render json: {
-          conversation: conversation_as_json(conversation)
-        }, status: :created
+        render json: conversation_as_json(conversation), status: :created
+      rescue ActiveRecord::RecordInvalid, ActionController::ParameterMissing, ActiveRecord::RecordNotFound
+        render json: I18n.t("api.endpoint_errors.conversations.cant_process"), status: :unprocessable_entity
```

Comment:
I don't see why you do that at the controller function level. It can be done at the Controller level.

Reply:
That endpoint is the only one that responds with that message and error code on a processing error according to the spec.  The rest only respond with 404 type errors.  It therefore seemed more appropriate to have it on the method instead of universal.